frontend/src/modules/history/components/Translation.js

```diff
@@ -99,8 +118,50 @@ export default class Translation extends React.Component<Props> {
         </a>
     }
 
+    toggleDiff = () => {
+        this.setState((state) => {
+            return { isDiffVisible: !state.isDiffVisible };
+        });
+    }
+
+    getDiff(base: string, target: string) {
+        const diff = dmp.diff_main(base, target);
+
+        dmp.diff_cleanupSemantic(diff);
+        dmp.diff_cleanupEfficiency(diff);
+
+        return diff;
+    }
+
+    renderDiff(translation: DBTranslation, activeTranslation: DBTranslation) {
+        const diff = this.getDiff(activeTranslation.string, translation.string)
+
+        return diff.map((item, index) => {
+            let type = item[0];
+            let slice = item[1];
+
+            switch(type) {
+                case DiffMatchPatch.DIFF_INSERT:
+                    return <ins key={ index }>{ slice }</ins>;
+
+                case DiffMatchPatch.DIFF_DELETE:
+                    return <del key={ index }>{ slice }</del>;
+
+                default:
+                    return <span key={ index }>{ slice }</span>;
+            }
+        });
+    }
```

Comment:
These `getDIff` and `renderDiff` function are not strictly related to this component, and I assume we might want to reuse that code in the Machinery tab, too. How would you feel about putting them somewhere else? Maybe it could be a component somewhere in `core`? Then below you would either render the simple string or the component.